SimplenoteUITests/Generic.swift

```diff
@@ -41,41 +41,43 @@ func getToAllNotes() {
 
 class Table {
 
+    class func getAllCells() -> XCUIElementQuery {
+        return app.tables.element.children(matching: .cell)
+    }
+
     class func getVisibleLabelledCellsNumber() -> Int {
         // We need only the table cells that have X = 0 and a non-empty label
         // otherwise we will include invisible elements from Sidebar pane, when Notes List is open
         // or the elements from Notes List when Settings are open
         // or the visible tags search suggestions
-        let cellsNum = app.tables.element.children(matching: .cell).count
-        var matchesNum: Int = 0
-        guard cellsNum > 0 else { return matchesNum }
+        let cells = Table.getAllCells()
+        var matches = 0
+        guard cells.count > 0 else { return matches }
 
-        for index in 0...cellsNum - 1 {
-            let cell = app.tables.cells.element(boundBy: index)
+        for cell in cells {
             if cell.frame.minX == 0.0 && cell.label.count > 0 {
-                matchesNum += 1
+                matches += 1
             }
         }
 
-        return matchesNum
+        return matches
```

Comment:
Nitpick. I'm fond of using collections manipulations instead of for loops.

Here's an alternative way of writing this code:

```swift
return Table.getAllCells()
    .filter { $0.frame.minX == 0.0 && $0.label.isEmpty == false }
    .count
```

I did not run the tests with it, so it might need some tweaking, but the idea is to use filter to get only the cells that match the criteria defined in your comment and return the count. If no cell matches, the count will be 0.

The advantage of this approach, in my opinion, is that there are no variables in the code, so less chances for mutable state to creep in and result in bugs. There's also less code to read, which is, _usually_, a good thing.

Reply:
This is a beautiful improvement @mokagio ! Thank you!